samples/contrib/kubeflow-katib/mpi-job-horovod.py

```diff
@@ -0,0 +1,235 @@
+                                    ],
+                                    "resources": {
+                                        "limits": {
+                                            "cpu": "500m",
+                                            "memory": "2Gi"
+                                        }
+                                    }
+                                }
+                            ]
+                        }
+                    }
+                },
+                "Worker": {
+                    "replicas": 2,
+                    "template": {
+                        "metadata": {
+                            "annotations": {
+                                "sidecar.istio.io/inject": "false"
+                            }
+                        },
+                        "spec": {
+                            "containers": [
+                                {
+                                    "image": "docker.io/kubeflow/mpi-horovod-mnist",
+                                    "name": "mpi-worker",
+                                    "resources": {
+                                        "limits": {
+                                            "cpu": "500m",
+                                            "memory": "4Gi"
+                                        }
+                                    }
+                                }
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    # Configure parameters for the Trial template.
+    trial_template = V1beta1TrialTemplate(
+        primary_pod_labels={
+            "mpi-job-role": "launcher"
+        },
+        primary_container_name="mpi-launcher",
+        success_condition='status.conditions.#(type=="Succeeded")#|#(status=="True")#',
+        failure_condition='status.conditions.#(type=="Failed")#|#(status=="True")#',
+        trial_parameters=[
+            V1beta1TrialParameterSpec(
+                name="learningRate",
+                description="Learning rate for the training model",
+                reference="lr"
+            ),
+            V1beta1TrialParameterSpec(
+                name="numberSteps",
+                description="Number of training steps",
+                reference="num-steps"
+            ),
+        ],
+        trial_spec=trial_spec
+    )
+
+    # Create Experiment specification.
+    experiment_spec = V1beta1ExperimentSpec(
+        max_trial_count=max_trial_count,
+        max_failed_trial_count=max_failed_trial_count,
+        parallel_trial_count=parallel_trial_count,
+        objective=objective,
+        algorithm=algorithm,
+        parameters=parameters,
+        trial_template=trial_template
+    )
+
+    output_file = "/output.txt"
+
+    # Katib launcher component.
+    # We convert Experiment spec to JSON dict.
+    # The Experiment is deleted after the Pipeline is finished.
+    op = dsl.ContainerOp(
```

Comment:
We suggest not to construct a `ContainerOp` instances directly. In fact, you should see a warning if you use it in this way. Can you please use an alternative?

https://github.com/kubeflow/pipelines/blob/60cf921c235b4b62f503b0ef79b47f79e30afeeb/sdk/python/kfp/dsl/_container_op.py#L1022-L1027

Reply:
Sure @chensun.
I will follow the same way as in the Notebook example.

Reply:
Done.